### ISSUE TITLE
Analysis should not run in referencing projects.

### ISSUE THREAD

**Issue by ItsSmiffy** [dotnet/roslyn#37107]
I have two projects, A and A.Tests.
A.Tests references A.
I have added fxcopanalyzers via NuGet into project A.
The analyzers are being pulled in and ran on the A.Tests project.

I have added `<PrivateAssets>all</PrivateAssets>` in the PackageReference for FxCopAnalyzers as mentioned in #24732 but hasn't resolved the issue.

n.b. Project A is a .NET Standard 2.0 project, Project A.Tests is a .NET Core.

**Comment by vatsalyaagrawal** [comment]
@mavasani for initial invesitgation.

**Comment by mavasani** [comment]
This indeed looks like a duplicate of #24732 which has been duped to NuGet/Home#4125. @ItsSmiffy did you try to close/re-open solution? Can you provide a repro project?

**Comment by ItsSmiffy** [comment]
@mavasani yes, I closed/re-opened the solution which didn't seem to resolve the issue. 
When creating a sample repo I could not reproduce the issue.
So going back to my original solution/projects I found that removing the project reference (which consumes the analyzers) from the Test project then re-adding the project to the Test project seems to ensure the Analyzers do not flow down to the Test project.

**Comment by CyrusNajmabadi** [comment]
I can't repro this.  And it appears as if it's a dupe of a closed issue.  closing out.
